### PR TITLE
openapi v3

### PR DESCRIPTION
**Pull request by adam-collins**
- switch to openapi v3
- make some /api/autocomplete parameters optional
- minor changes

### CODE REVIEW

File: server/src/main/java/au/org/ala/names/ws/ALANameMatchingServiceConfiguration.java [hunk 1]
```diff
@@ -33,13 +49,30 @@ public class ALANameMatchingServiceConfiguration extends Configuration {
...
-        this.swagger.setVersion("1.8-SNAPSHOT");
-        this.swagger.getSwaggerViewConfiguration().setPageTitle("ALA Namematching API");
+        // swagger UI
+        this.swaggerBundleConfiguration.setTitle("ALA Namematching API");
+        this.swaggerBundleConfiguration.setDescription("A taxonomy service that maps scientific name queries onto taxon concepts");
+        this.swaggerBundleConfiguration.setContactUrl("https://ala.org.au");
+        this.swaggerBundleConfiguration.setContactEmail("support@ala.org.au");
+        this.swaggerBundleConfiguration.setResourcePackage("au.org.ala.names.ws.api,au.org.ala.names.ws.client,au.org.ala.names.ws.resources");
+        this.swaggerBundleConfiguration.setLicense("Mozilla Public Licence 1.1");
+        this.swaggerBundleConfiguration.setVersion("1.8-SNAPSHOT");
```
Comment (djtfmartin) [inline server/src/main/java/au/org/ala/names/ws/ALANameMatchingServiceConfiguration.java:59]: Is this correct ?

**Review by djtfmartin — APPROVED** [review]
Approved

Just a minor issue with
```
this.swaggerBundleConfiguration.setVersion("1.8-SNAPSHOT");
```

File: server/src/main/java/au/org/ala/names/ws/ALANameMatchingServiceConfiguration.java (hunk 1 above)
Comment (charvolant) [reply server/src/main/java/au/org/ala/names/ws/ALANameMatchingServiceConfiguration.java:59]: Looks right to me. It's just a renaming of the bundle for clarity.

**Review by charvolant — APPROVED** [review]

File: server/src/main/java/au/org/ala/names/ws/ALANameMatchingServiceConfiguration.java (hunk 1 above)
Comment (djtfmartin) [reply server/src/main/java/au/org/ala/names/ws/ALANameMatchingServiceConfiguration.java:59]: I specifically meant the "1.8-SNAPSHOT" bit

**Pull request merged**